### ISSUE TITLE
"email is already in use" error on email verification and spinner of death

### ISSUE THREAD

**Issue by jaywink** [vector-im/element-web#15916]
### Description

Registering to a new host tied to vector.im IS with an existing email shows server error (as per #13767), but also crashes the registration flow with a spinner of death.

![Selection_900](https://user-images.githubusercontent.com/1174866/101463788-8a8e2f00-3946-11eb-9473-387c64634e27.png)

Reloading the page logged me in. The error seems to indicate the registration of my account failed when in reality only the binding to the IS failed.

This might make the user think the registration did fail and just close the browser window OR re-register another account, thus contributing to a bad user experience. This is especially bad for EMS since many users using matrix.org might try an EMS host and are likely to use the same email.

### Steps to reproduce

- Bind an email address with vector.im
- Try to register on another host with the same email

![Selection_897](https://user-images.githubusercontent.com/1174866/101464700-a7773200-3947-11eb-8e47-6041b85e8a40.png)

### Version information

- **Platform**: web
- **Browser**: Firefox
- **OS**: Ubuntu
- **URL**: EMS hosted element test host v1.7.14

**Comment by jryans** [comment]
Hmm, so far I can't reproduce this issue. As best as I can tell, @jaywink is saying this is about emails bound to the IS (not the HS).

So far in my testing:

* If I create a new account using an email already known to the HS (using matrix.org as the test HS), I do get a registration error about "email already in use" (but as far as I can tell, that's not what @jaywink is reporting here)
* If I create a new account using an email already known to IS but not the HS (using matrix.org as the test HS), registration complete successfully

The test results above make sense to me: after the 2019 privacy changes, we should not be communicating with the IS directly during registration, so we would only know about the HS state.

**Comment by jryans** [comment]
Also, so far I have not seen the registration form hang on a spinner in any case so far: when I did receive the email in use error, the form appeared and I was free to try again.

**Comment by jryans** [comment]
It sounds like @jaywink cannot reproduce this either now, so let's assume it's working until we hear more.

**Comment by jaywink** [comment]
Thanks @jryans for checking this. Indeed, as indicated, the newly registered account is not bound to the IS at this stage. I made the mistake of assuming so when logging the issue.

Unfortunately I can't reproduce this exact behaviour with a new host or even with the old one. I did manage to reproduce a spinner of death if clicking the verification email twice quickly, but that is a separate issue which I couldn't reproduce afterwards either. Possibly hit some really odd race condition.
